kie-soup-maven-utils/kie-soup-maven-integration/src/main/java/org/appformer/maven/integration/MavenRepositoryConfiguration.java

```diff
@@ -205,12 +213,24 @@ private ArtifactRepository toArtifactRepository( RemoteRepository remoteReposito
             artifactRepository.setAuthentication( new Authentication( server.getUsername(),
                                                                       server.getPassword() ) );
         }
-        if (settings.getActiveProxy() != null) {
-            artifactRepository.setProxy(getActiveMavenProxyFromSettings(settings));
+
+        if ( activeProxy != null ) {
+            if ( null == activeProxy.getNonProxyHosts() ) {
+                artifactRepository.setProxy( getActiveMavenProxyFromSettings( settings ) );
+            } else if( ! addProxy( settings.getActiveProxy().getNonProxyHosts(), artifactRepository.getUrl() ) ) {
+                artifactRepository.setProxy( getActiveMavenProxyFromSettings( settings) );
+            }
         }
+
         return artifactRepository;
     }
 
+    private static boolean addProxy ( String nonProxyHosts, String artifactURL ) {
+        Pattern p = Pattern.compile( nonProxyHosts );
+        Matcher m = p.matcher( artifactURL );
```

Comment:
The pattern is matching against whole URL. IMHO we should match only against hostname.